torch/csrc/inductor/aoti_torch/oss_proxy_executor.cpp

```diff
@@ -0,0 +1,266 @@
+#include <nlohmann/json.hpp>
+#include <fstream>
+#include <iostream>
+
+#include <torch/csrc/inductor/aoti_torch/oss_proxy_executor.h>
+
+using json = nlohmann::json;
+
+namespace {
+at::Tensor* tensor_handle_to_tensor_pointer(AtenTensorHandle handle) {
+  return reinterpret_cast<at::Tensor*>(handle);
+}
+} // namespace
+
+namespace torch {
+namespace aot_inductor {
+
+void OSSProxyExecutor::prefill_stack_with_static_arguments(
+    int index,
+    at::TypePtr schema_arg_type,
+    json serialized_arg,
+    OpKernel& op_kernel) {
+  auto& stack = op_kernel.stack_;
+  auto& dynamic_args = op_kernel.dynamic_args_;
+
+  TORCH_CHECK(serialized_arg.size() == 1);
+  std::string serialized_arg_type = serialized_arg.begin().key();
+  auto& serialized_arg_val = serialized_arg.begin().value();
+
+  switch (schema_arg_type->kind()) {
+    case c10::TypeKind::TensorType: {
+      TORCH_CHECK(serialized_arg_type == "as_tensor");
+      stack.emplace_back();
+      dynamic_args.emplace_back(
+          index, DynamicArgType::TensorType, 1, serialized_arg_val);
```

Comment:
missing std::move(serialized_arg_val)

more importantly, IIUC we are going to parse JSON once at startup and then use it in a potentially-performance-critical path, potentially very many times. Shouldn't we extract the data into plain old structs to save memory and hash-table-lookups later?

Reply:
Ah yes that makes sense! I can address this in a follow up!